contracts/DivviRegistry.sol

```diff
@@ -37,6 +60,21 @@ contract DivviRegistry is
     address indexed rewardsProvider,
     address indexed rewardsConsumer
   );
+  event ReferralRegistered(
+    address user,
+    address indexed rewardsProvider,
+    address indexed rewardsConsumer,
+    uint256 indexed chainId,
+    bytes32 txHash
+  );
+  event ReferralSkipped(
```

Comment:
my understanding of this "indexed" keyword is that it makes it faster to query the blockchain for events with this property, but i'm not really sure how we might want to analyse these events. there can only be maximum 3 indexed properties per event, so i've chosen to be able to quickly see how often a consumer is unsuccessful in the referral / how often a user and tx hash is bouncing. this info would help to raise awareness of bugs maybe. open for alternative preferences for which properties are indexed

Reply:
Hmm I would probably advocate for indexing `user` `rewardsProvider` and `rewardsConsumer`, since this lets us segment events across each of the main parties we care about in the protocol. e.g., we can easily see:
* All the entities a user is referred to
* All the users referred to a particular entity
* All the users referred by a particular entity

If we want to look up by `txHash`, and we know the user who submitted the `txHash` (which we always should, since we can just check the sender of the transaction), we can just look up events by user (should be a relatively small number per user) and filter them down.

Reply:
that makes sense, thanks for the reasoning! updated!

Reply:
i also updated ReferralRegistered to have the same for consistency, i hope this is expected